### DISCUSSION TITLE
Sorting by date

### DISCUSSION THREAD

**Discussion by DannyHatcher-Git** [RafaelGB/obsidian-db-folder#475] [Q&A] [answered]
When sorting by date in Ascending order it puts the files with no date at the top and then goes in the order I want.
Is there a way to push those files to the bottom of the list without adding a date?

My use case is a project list sorted by due date with the closest deadline at the top.

**Comment by RafaelGB** [comment, marked as answer]
currently you can use a filter with that column using the operator `is not empty` so your raws will appear in order without those cases. You can use also a filter group to quickly enable/disable the option

**Reply by DannyHatcher-Git** [reply]
Yeah that is the only solution I have at the moment but it is irritating I have to go in and out of the filters 🤷‍♂️
My preference would be for the sort to be active on `not empty` cells and then leave the others at the bottom. 

Is that doable?

**Reply by RafaelGB** [reply]
I am afraid its not without refactor the sort function. The plugins parse all data to text before sort them and the empty info is considered as `""`
The `react-table` framework control how is sorted but its API allow you to create your own function. The function does not have access to the direction (asc or desc) so I can not apply logic for empty cells in function of that direction and keep them on bottom (or on top) with some config 

```ts
const dbfolderColumnSortingFn: (
  ddbbConfig: LocalSettings
) => SortingFn<RowDataType> =
  (ddbbConfig: LocalSettings) =>
  (
    rowA: Row<RowDataType>,
    rowB: Row<RowDataType>,
    columnId: string
  ): number => {
    const a = ParseService.parseLiteral(
      rowA.getValue<Literal>(columnId),
      InputType.MARKDOWN,
      ddbbConfig,
      true
    )
      .toString()
      .toLowerCase();
    const b = ParseService.parseLiteral(
      rowB.getValue<Literal>(columnId),
      InputType.MARKDOWN,
      ddbbConfig,
      true
    )
      .toString()
      .toLowerCase();
    return a === b ? 0 : a > b ? 1 : -1;
  };

export default dbfolderColumnSortingFn;
```

**Reply by DannyHatcher-Git** [reply]
I will look over the code to try and understand it as I am learning but thank you for explaining it to me.

**Comment by DryIce1** [comment]
Hi @DannyHatcher-Git , 
i'm just setting up Folder DB for the first time. Can you help me set up a workflow please.

What do you currently use for sorting your daily notes by date? 
- Do I need to add a new piece of metadata to my daily notes and sort by that? 
  - For example, I could use the Linter plugin to create a 'date' YAML metadata on all my daily notes and insert the daily note's title (which is in the yyyy-MM-dd format).
- Or can I refine the FROM search terms in Folder DB to select files from FOLDER "Logs folder" and TAG "#dailynotes"?

thanks for your help.
